src/topology/category/TopCommRing.lean

```diff
@@ -35,8 +38,12 @@ instance : concrete_category TopCommRing.{u} :=
         dsimp, apply continuous.comp ; assumption
       end⟩ },
   forget := { obj := λ R, R, map := λ R S f, f.val },
-  forget_faithful := { }
-}
+  forget_faithful := { } }
+end
+
+instance (X : TopCommRing) : comm_ring X := X.is_comm_ring
+instance (X : TopCommRing) : topological_space X := X.is_topological_space
+instance (X : TopCommRing) : topological_ring X := X.is_topological_ring
```

Comment:
What is the difference between `instance (X : TopCommRing) : topological_ring X := X.is_topological_ring` and `attribute [instance] TopCommRing.is_topological_ring`?

Reply:
I think there's a difference: these instances are for, e.g. `comm_ring (X : Type u)`, whereas `TopCommRing.is_comm_ring : comm_ring (X.α)`. These are definitionally but not syntactically equal, so it makes a difference in typeclass search unfortunately.